gdx/src/com/badlogic/gdx/FadingGame.java

```diff
@@ -0,0 +1,199 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.graphics.g2d.Batch;
+import com.badlogic.gdx.graphics.glutils.FrameBuffer;
+import com.badlogic.gdx.utils.transition.ScreenTransition;
+import com.badlogic.gdx.utils.transition.TransitionListener;
+
+/** An {@link Game} that delegates to a {@link Screen}. Allows to apply different transitions to screens. </p>
+ * <p>
+ * Screens are not disposed automatically. You must handle whether you want to keep screens around or dispose of them when another
+ * screen is set.
+ * @author iXeption */
+public class FadingGame extends Game {
+
+	protected FrameBuffer currentScreenFBO;
+	protected FrameBuffer nextScreenFBO;
+	protected Screen nextScreen;
+	protected Batch batch;
+
+	private List<TransitionListener> listerns;
+
+	private float transitionDuration;
+	private float currentTransitionTime;
+	private boolean transitionRunning;
+	private ScreenTransition screenTransition;
+
+	public FadingGame (Batch batch) {
+		this.batch = batch;
+	}
+
+	@Override
+	public void create () {
+		currentScreenFBO = new FrameBuffer(Pixmap.Format.RGBA8888, Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), false);
+		nextScreenFBO = new FrameBuffer(Pixmap.Format.RGBA8888, Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), false);
+		listerns = new ArrayList<TransitionListener>();
+	}
+
+	@Override
+	public void dispose () {
+		if (screen != null) screen.hide();
+		if (nextScreen != null) nextScreen.hide();
+
+	}
+
+	@Override
+	public void pause () {
+		if (screen != null) screen.pause();
+		if (nextScreen != null) nextScreen.pause();
+	}
+
+	@Override
+	public void resume () {
+		if (screen != null) screen.resume();
+		if (nextScreen != null) nextScreen.resume();
+	}
+
+	@Override
+	public void render () {
+
+		float delta = Gdx.graphics.getDeltaTime();
+
+		if (nextScreen == null) {
+			// no other screen
+			screen.render(delta);
+		} else {
+			if (transitionRunning && currentTransitionTime >= transitionDuration) {
+				// is active and time limit reached
+				this.screen.hide();
+				this.screen = this.nextScreen;
+				this.screen.resume();
+				transitionRunning = false;
+				this.nextScreen = null;
+				notifyFinished();
+
+			} else {
+				// transition is active
+				if (screenTransition != null) {
+					currentScreenFBO.begin();
+					this.screen.render(delta);
+					currentScreenFBO.end();
+
+					nextScreenFBO.begin();
+					this.nextScreen.render(delta);
+					nextScreenFBO.end();
+
+					float percent = currentTransitionTime / transitionDuration;
+
+					screenTransition.render(batch, currentScreenFBO.getColorBufferTexture(), nextScreenFBO.getColorBufferTexture(),
+						percent);
+					currentTransitionTime += delta;
+
+				}
+
+			}
+
+		}
+
+	}
+
+	@Override
+	public void resize (int width, int height) {				
+		if (screen != null) screen.resize(width, height);
+		if (nextScreen != null) nextScreen.resize(width, height);
```

Comment:
Unless I am missing something, currentScreenFBO and nextScreenFBO should be resized too. I suggest making that a separate protected method to allow customized resizing, since creating them can already be customized and FBOs are heavy objects.